### ISSUE TITLE
Text selection for file paths not working when path ends with a colon followed by text

### ISSUE THREAD

**Issue by emerick** [mintty/mintty#1236]
In the following example:

```
$ git grep kExtensionTelemetryTabsApiSignal
chrome/browser/extensions/api/tabs/tabs_api.cc:          safe_browsing::kExtensionTelemetryTabsApiSignal)) {
chrome/browser/safe_browsing/extension_telemetry/extension_telemetry_service_browsertest.cc:         kExtensionTelemetryTabsApiSignal},
components/safe_browsing/core/common/features.cc:BASE_FEATURE(kExtensionTelemetryTabsApiSignal,
components/safe_browsing/core/common/features.cc:    {&kExtensionTelemetryTabsApiSignal, true},
components/safe_browsing/core/common/features.h:BASE_DECLARE_FEATURE(kExtensionTelemetryTabsApiSignal);
```

If I double-click the path in the first result from git grep above, the appropriate text is selected:
In other words, **chrome/browser/extensions/api/tabs/tabs_api.cc** is selected.

If I double-click the path in the third result from git grep above, the colon and the first word following the colon is also selected.
In other words, **components/safe_browsing/core/common/features.cc:BASE_FEATURE** is selected. The selection should have stopped before the colon.

This is new (and I think undesirable) behavior that I first noticed in mintty 3.6.4.

**Comment by ak2** [comment]
The fix in 8f538570 is working nicely for me. The special handling of `:/` and `::` looks a good compromise between the use case here and the ones for URLs, IPv6 addresses and namespace qualifiers.

**Comment by mintty** [comment]
Released 3.7.0.
